File: be/Models/Street.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using NetTopologySuite.Geometries;

namespace be.Models
{
    public class Street
    {
        [Key]
        public int Id {  get; set; }
        [Required]
        public string StreetName { get; set; } = "";
        public string Address { get; set; } = "";
        public string Description { get; set; } = "";
        public string ImageUrl { get; set; } = "";
        public DateTime UpdatedDate { get; set; }
        public DateTime CreatedDate { get; set; }
        public LineString? Route { get; set; }
        public LineString? WayPoints { get; set; }
        public LineString? ManualRoute { get; set; }
        public LineString? ManualWayPoints { get; set; }
        public bool IsApproved { get; set; }
        public List<StreetHistory> Histories { get; set; } = new List<StreetHistory>();
        public List<StreetImage> Images { get; set; } = new List<StreetImage>();
        public required int StreetTypeId { get; set; }
        public StreetType? StreetType { get; set; }
    }
}

File: fe_admin/src/types/street.ts
export interface IStreetListContext {
  filter: string;
  setFilter: (filter: string) => void;
  sortBy: string;
  setSortBy: (sortBy: string) => void;
  isDesc: boolean;
  setIsDesc: (isDesc: boolean) => void;
  pageNumber: number;
  setPageNumber: (pageNumber: number) => void;
  pageSize: number;
  setPageSize: (pageSize: number) => void;
}

export interface IStreetSearchList {
  streets: IStreetSearch[];
  totalPages: number;
}

export interface IStreetSearch {
  id: number;
  streetName: string;
  streetType: IStreetType;
  address: string;
  description: string;
  imageUrl: string;
  updatedDate: string; // You can use Date if you plan to parse it into a Date object
  createdDate: string; // You can use Date if you plan to parse it into a Date object
  isApproved: boolean;
}

export interface IStreet {
  id: number;
  streetName: string;
  streetTypeId: number;
  address: string;
  description: string;
  imageUrl: string;
  isApproved: boolean;
  weight: number;
  opacity: number;
  color: string;
  images: IStreetImage[];
  histories: IStreetHistory[];
  updatedDate: string; // You can use Date if you plan to parse it into a Date object
  createdDate: string; // You can use Date if you plan to parse it into a Date object
  wayPoints: {
    coordinates: [number, number][];
  };
  route: {
    coordinates: [number, number][];
  };
  manualWayPoints: {
    coordinates: [number, number][];
  };
  manualRoute: {
    coordinates: [number, number][];
  };
}

export interface IStreetType {
  id: number;
  streetTypeName: string;
  createdDate?: string;
  updatedDate?: string;
}

export interface IStreetTypeList {
  streetTypes: IStreetType[];
  totalPages?: number;
}

export interface ISelectOption {
  value: number | string;
  label: string;
}

export interface IStreetImage {
  imageUrl?: string;
  publicId?: string;
  description?: string;
}

export interface IStreetHistory {
  id: string;
  period: string;
  description: string;
}
